### ISSUE TITLE
hcat, vcat, hvcat, and cat

### ISSUE THREAD

**Issue by schmrlng** [SimonDanisch/FixedSizeArrays.jl#52]
When using [ImmutableArrays.jl](https://github.com/JuliaGeometry/ImmutableArrays.jl), the fact that everything is a subtype of Base.AbstractArray makes the various concatenation operations work automatically (promoting everything to a dense array, which seems reasonable when you're already growing array sizes). On the roadmap it looks like inheriting from AbstractArray is an eventual goal; do you have an idea of what work needs to get done to make that happen? If it seems pretty far off, in the mean time would it make sense to PR in some hacks (also for slicing, etc.)?

**Comment by SimonDanisch** [comment]
Yes, it is relatively far off, mainly due to time constraints and priorities :(
The issues to look out for are: JuliaLang/LinearAlgebra.jl#255, JuliaLang/julia#11610 and JuliaLang/julia#11902
The first two are somewhat interchangable, but at least one of them needs to be resolved. The last is just helpful ;)
If you have some time and want to play around with slices and stuff like that, feel free to open PR's!
If things are not documented well enough, feel free to ask any questions here or on gitter!

Best,
Simon

**Comment by c42f** [comment]
Just a note that I started looking at this (see the first commit on the concatenation branch which contains a partial implementation), but may not get back to it for a little while.
